### PR TITLE
backport v10:  Add docs for EC2 instance discovery #16279 

### PR DESCRIPTION
**Pull request by lxea**
Original: #16279

### CODE REVIEW

**Review by ptgott — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Review by r0mant — COMMENTED** [review]
@lxea We should mention somewhere in this guide that EC2 auto-discovery feature will not handle agent upgrades and users will be responsible for upgrading already enrolled nodes themselves until we implement auto-upgrades.

File: docs/config.json [hunk 1]
```diff
@@ -603,6 +603,10 @@
             {
               "title": "Host User Creation",
               "slug": "/server-access/guides/host-user-creation/"
+            },
+            {
+              "title": "EC2 Instance Discovery",
```
Comment (r0mant) [inline docs/config.json:608]: @lxea This should also be marked as a preview feature.
```suggestion
              "title": "EC2 Instance Discovery (Preview)",
```

File: docs/pages/server-access/guides/ec2-discovery.mdx [hunk 2]
```diff
@@ -0,0 +1,313 @@
+---
+title: Configure Teleport to Automatically Enroll EC2 instances
```
Comment (r0mant) [inline docs/pages/server-access/guides/ec2-discovery.mdx:2]:
```suggestion
title: Configure Teleport to Automatically Enroll EC2 instances (Preview)
```

File: docs/pages/server-access/guides.mdx [hunk 3]
```diff
@@ -11,3 +11,4 @@ layout: tocless-doc
 - [Restricted Session](./guides/restricted-session.mdx): How to configure and use Restricted Session to apply security policies to SSH sessions.
 - [Visual Studio Code](./guides/vscode.mdx): How to remotely develop with Visual Studio Code and Teleport.
 - [Host User Creation](./guides/host-user-creation.mdx): How to configure Teleport to automatically create transient host users.
+- [EC2 Instance Discovery](./guides/ec2-discovery.mdx): How to configure Teleport to automatically enroll EC2 instances.
```
Comment (r0mant) [inline docs/pages/server-access/guides.mdx:14]:
```suggestion
- [EC2 Instance Discovery (Preview)](./guides/ec2-discovery.mdx): How to configure Teleport to automatically enroll EC2 instances.
```

**Review by r0mant — COMMENTED** [review]
@lxea Please forward-port all comments to v11 and master as well.

**Pull request merged**